### DISCUSSION TITLE
Pivot Grid shows no data

### DISCUSSION THREAD

**Discussion by georgemaka** [TabularEditor/TabularEditor3#1140] [🙋♀️TE3 Q&A] [answered]
Why does pivot grid not show any data? The same columns/rows that I used in Power BI are the same that I am using in Tabular Editor. Power BI is showing the values, but nothing in Tabular Editor. 

Also, tabular editor does not allow to add the data values of a column? Can I only add a measure for data?

**Comment by otykier** [comment, marked as answer]
Hi @georgemaka 

> Why does pivot grid not show any data?

The Pivot Grid in TE3 is based on MDX. Could you please check that any model columns that you add to the columns/rows/filter area of the Pivot Grid does not have the "IsAvailableInMdx" property set to false? Also, please be aware that the Pivot Grid does not display any data until at least one measure has been added to the data area.

> Can I only add a measure for data?

Yes - this is unfortunately a limitation of MDX - it only works with explicit measures. You would see the same limitation if you tried to browse your data model using a Pivot Table in Excel. For this and a few other reasons, it is a best practice to always create explicit measures in your model. You can [use a c# script to quickly create these measures](https://docs.tabulareditor.com/te2/Useful-script-snippets.html#create-measures-from-columns) if needed.
